app/src/main/java/de/xikolo/utils/extensions/TextViewExtensions.kt

```
@@ -1,162 +1,98 @@
-
-        val result = BitmapDrawablePlaceHolder()
-
-        object : AsyncTask<Void, Void, Bitmap>() {
-
-            override fun doInBackground(vararg meh: Void): Bitmap? {
-                var uri = Uri.parse(source)
-                if (uri.isRelative) {
-                    uri = Uri.Builder()
-                        .scheme("https")
-                        .authority(Config.HOST)
-                        .path(uri.path)
-                        .query(uri.query)
-                        .build()
-                }
-                try {
-                    return glideRequests
-                        .asBitmap()
-                        .load(uri)
-                        .centerCrop()
-                        .noPlaceholders()
-                        .dontAnimate()
-                        .submit()
-                        .get()
-                } catch (e: Exception) {
-                    Log.e(TAG, e.message, e)
-                    return null
+        fun getAbsoluteUrl(url: String): String {
+            try {
+                val uri = URI(url)
+                if (!uri.isAbsolute) {
+                    return URL(URL(Config.HOST_URL), url).toString()
                 }
-
+            } catch (e: URISyntaxException) {
+                return url
+            } catch (e: MalformedURLException) {
+                return url
             }
+            return url
+        }
 
-            override fun onPostExecute(bitmap: Bitmap) {
-                val textView = textViewWeakReference.get() ?: return
-                try {
-                    if (maxWidth == -1) {
-                        val horizontalPadding = textView.paddingLeft + textView.paddingRight
-                        maxWidth = textView.measuredWidth - horizontalPadding
-                        if (maxWidth == 0) {
-                            maxWidth = Integer.MAX_VALUE
-                        }
+        val glide = Glide.with(context)
```

This way it uses the same configured `glide` that we use at other places, right?

However, I would move this into the plugin methods like in the [example](https://noties.io/Markwon/docs/v4/image-glide/).

---

I had it that way before and it crashed sometimes. If you look at the default implementation of `GlideImagesPlugin.create(...)` [here](https://github.com/noties/Markwon/blob/master/markwon-image-glide/src/main/java/io/noties/markwon/image/glide/GlideImagesPlugin.java), you see this hint:

```
@NonNull
    public static GlideImagesPlugin create(@NonNull final Context context) {
        // @since 4.5.0 cache RequestManager
        //  sometimes `cancel` would be called after activity is destroyed,
        //  so `Glide.with(context)` will throw an exception
        return create(Glide.with(context));
    }
```

---

But wouldn't it be cleaner to check if the context is still valid beforehand? 🤔  (If this is possible.)

---

we could place everything inside a try-catch block but I would go with the default implementation

---

It could also be placed inside the interface implementation object

---

> It could also be placed inside the interface implementation object

I made a commit for this. I think it achieves your requirement

---

Silently swallowing exceptions is also not so nice. I would prefer to avoid this by checking the state of context before.

---

But checking the context validity would require to look into what Glide specifies as "valid", which might change between versions

---

Who throws the exception based on which condition? As far as I understand this happens when retrieving the Glide Singleton, right? And not during the `clear` operation? Because the latter would still have this problem now.

---

The problem is that `clear` is sometimes called when the activity has already been destroyed and `Glide.with(context)` fails.
When placing the `Glide.with(context)` inside the `clear` method, the app will sometimes crash. But if it is placed outside, e.g. in the object's scope, the `glide` variable is initialized when the object is initialized (and the activity is running), leaving a working Glide instance in `glide` to even use in `clear` when `Glide.with(context)` would fail, because it has been cached.

Is it that what you mean?

---

Yup, I guess I'm fine with it. Having the glide retrieval inside the anonymous plugin object is much cleaner.